src/System.IO.Pipelines/MemoryEnumerator.cs

```diff
@@ -17,7 +17,7 @@ public struct MemoryEnumerator
         public MemoryEnumerator(ReadCursor start, ReadCursor end)
         {
             _segmentEnumerator = new SegmentEnumerator(start, end);
-            _current = Memory<byte>.Empty;
+            _current = default(Memory<byte>);
```

Comment:
It's `default(Memory<byte>)` before you call `MoveNext` and after it returns false, and for enumerator it's undefined behavior.